File: lab/5-6/lab_5-6.py
def one():
    from random import  randint
    A = [randint(-99, 67) for i in range(30)]
    print(f'Массив A: {A}')
    print(f'Массив B: {[i for i in A if i % 2 == 0]}')


def two():
    n = int(input('Размер массива: '))
    print('Элементы массива через пробел:')
    arr = list(map(int, input().split()))
    print(arr == sorted(arr))


def three():
    from random import randint
    arr = [randint(-50, 100) for i in range(20)]
    set(arr); list(arr)
    print(f'Массив: {arr}')

    arr_copy = arr.copy()
    for i in reversed(arr_copy):
        if i > 0:
            arr_copy.remove(i)
            print(f'Без последнего положительного элемента ({i}): {arr_copy}')
            break
    
    arr_copy = arr.copy()
    max_value = max(arr_copy)
    arr_copy.remove(max_value)
    print(f'Без наибольшего элемента ({max_value}): {arr_copy}')


one()
two()
three()

File: lab/5-6/lab_5-6.pas
function RemoveAt(var arr: array of integer; index: integer): array of integer;
begin
  var originalLength := arr.Length;
  for var i := index to originalLength - 2 do
    arr[i] := arr[i + 1];
  
  SetLength(arr, originalLength - 1);
  result := arr;
end;


procedure one();
begin
  var A := arrrandom(n := 30, a := -99, b := 67);
  var B := A.where(x -> x mod 2 = 0);
  writeln('Массив A: ', A);
  writeln('Массив B: ', B);
end;


procedure two();
begin
  var n := readinteger('Размер массива: ');
  writeln('Элементы массива: ');
  var arr := readarrinteger(n);
  println(arr.SequenceEqual(arr.sorted()));
end;


procedure three();
begin
  var arr := arrrandom(n := 20, a := -50, b := 100);
  println('Массив: ', arr);
  
  var index := -1;
  for var i := arr.High downto 0 do
  begin
    if arr[i] > 0 then
    begin
      index := i;
      arr := removeat(arr, index);
      break;
    end;
end;
  println('Без последнего положительного элемента', arr);
  arr := removeat(arr, arr.IndexMax());
  print('Без наибольшего элемента', arr);
end;


begin
  one();
  two();
  three();
end.